app/controllers/mentors_controller.rb

```diff
@@ -11,12 +11,18 @@ def index
 
     # mentors are filtered by the criteria above
     last_selected_coach = params[:mentor][:filter_by_coach_id]
+    last_selected_meeting_day = params[:mentor][:filter_by_meeting_day]
     if params[:mentor][:filter_by_coach_id].to_i > 0
       @mentors = @mentors.joins(:admins).where('kids.admin_id = ?', params[:mentor][:filter_by_coach_id].to_i).uniq
       params[:mentor][:filter_by_coach_id] = nil
     end
+    if params[:mentor][:filter_by_meeting_day].to_i > 0
+      @mentors = @mentors.joins(:kids).where('kids.meeting_day = ?', params[:mentor][:filter_by_meeting_day].to_i).uniq
```

Comment:
If i remove to_i from line 19 then it produces this error:
MentorsController as an admin index assigns two mentors in the index
     Failure/Error: if params[:mentor][:filter_by_meeting_day] > 0
     
     NoMethodError:
       undefined method `>' for nil:NilClass

to_i in line 20 is indeed not needed. The tests pass without it and the functionality is not affected. 

Uniq is needed as otherwise duplicate results are produced (confirmed by the tests).